Userland/Libraries/LibLine/Editor.cpp

```diff
@@ -2006,6 +2006,40 @@ VTState actual_rendered_string_length_step(StringMetrics& metrics, size_t index,
     return state;
 }
 
+static Result<char, Editor::Error> try_read_char_for_vt_dsr_with_retries()
+{
+    char c;
+    Editor::Error error;
+    bool read_success = false;
+
+    // Sometimes you have to be a little patient for terminal to start outputting
+    // the response for "Report cursor position". For example, this happens after
+    // Ctrl+C'ing the "top" command.
+    for (auto retries = 0; !read_success && retries < 5; ++retries) {
```

Comment:
pls rewrite this loop to be _a bit_ less MISRA compliant, it's okay, $BIG_EMBEDDED_CORP won't find out ;)

(i.e. just return as early as possible, move that variable up there into here, etc etc.